librz/core/rop.c

```diff
@@ -416,29 +444,62 @@ RZ_API bool rz_core_rop_gadget_info_has_register(const RZ_NONNULL RzRopGadgetInf
 	return false;
 }
 
-static bool is_var_write_event(const RzRopRegInfo *reg_info) {
+static inline bool is_var_read_event(const RzRopRegInfo *reg_info) {
+	return reg_info->is_var_read && !reg_info->is_mem_read && !reg_info->is_var_write && !reg_info->is_mem_write;
```

Comment:
why this one is checking also mem read & all the writes?

Reply:
This is being done on the older event which is being pushed onto the stack. It is essential for IDing the cases like these:
```
0x00401000: mov ebx, eax; ret;
```
Maybe I will reduce the complexity of this logic in future.